core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -190,11 +197,15 @@ private[spark] class TaskSetManager(
     }
 
     for (loc <- tasks(index).preferredLocations) {
+
+      val locHost: String = if( sparkLocalHostname != None ) loc.host
```

Comment:
Style issue: spaces outside parens, not inside. This could also be a `.getOrElse` construct maybe. Why does resolving it to an IP address depend on whether this host set `SPARK_LOCAL_HOSTNAME`?